docs/IntegrationGuide.md

```diff
@@ -1,95 +1,306 @@
+              If you need prevent address change by customers (buyers) - use `shippingPreference` below.
+
+            - `shippingPreference` (custom field): value of `Payment#application_context#shipping_preference` enum.
+            This value is used to allow/block shipping address change by the customer (buyer). Reed more at
+            [Application Context Documentation](https://developer.paypal.com/docs/api/orders/#definition-application_context)
+
+              **Note**: So far this feature is not properly documented by Paypal API developers, so the reference above
+              actually refers to [Orders API](https://developer.paypal.com/docs/api/orders/#definition-application_context),
+              instead of [Payments API](https://developer.paypal.com/docs/api/payments/#definition-application_context).
+              Respective issues are created:
+                - [Payment/Order#applicationContext property is not available](https://github.com/paypal/PayPal-Java-SDK/issues/330#issuecomment-356008914)
+                - [Payment application_context is not documented](https://github.com/paypal/PayPal-REST-API-issues/issues/179)
+
+    1. Backend POSTs CTP payment ID (created in the previous step) to Paypal-integration (see also [HTTP Responses](#http-responses)). Example:
+        ```
+        POST https://paypal-plus-integration-server.com/${tenantName}/commercetools/create/payments/${ctpPaymentId}
+        ```
+        If request was successful both response body as also CTP payment object (as custom field) will have `approvalUrl` defined.
+
+    1. Frontend uses returned `approvalUrl` to render available payment methods within PayPal hosted iFrame. For more details see official [Paypal Plus integration guide](https://www.paypalobjects.com/webstatic/de_DE/downloads/PayPal-PLUS-IntegrationGuide.pdf).
+
+1. **Submit user's addresses to Paypal Plus**
+
+    After user's payment method selection and before redirection of the user to Paypal, backend has to POST CTP payment ID to Paypal-integration (see also [HTTP Responses](#http-responses)):
+        ```
+        POST https://paypal-plus-integration-server.com/${tenantName}/commercetools/patch/payments/${ctpPaymentId}
+        ```
+    **Note**: Refer to the newest version of [Paypal Plus integration guide](https://www.paypalobjects.com/webstatic/de_DE/downloads/PayPal-PLUS-IntegrationGuide.pdf) in order to make a request
+    before redirect in Javascript. As of August 2017, on submit should call `ppp.doContinue()`. Additionally,
+    `ppp` object must be created with the following option:
+     ```javascript
+         var ppp = PAYPAL.apps.PPP({
+             onContinue: function () {
+                   $.post("/url-to-your-shop-that-will-make-call-to-paypal-integration", function (data) {
+                   if (data != false) {
+                     ppp.doCheckout();
+                   }
+               });
+             }
+         });
+     ```
```

Comment:
the whole paragraph above is a bit confusing. Maybe we have to refer explicitly smth like:

 - when `approvalUrl` is available - show Paypal Plus secure iframe to the buyer
 - Optionally, if shipping address should be specified on the created payment, do payment patch using:
```
POST https://paypal-plus-integration-server.com/${tenantName}/commercetools/patch/payments/${ctpPaymentId}
```
(and then continue with referring to PPP documentation and code sample)